### ISSUE TITLE
Implement Term through proxy path

### ISSUE THREAD

**Issue by RubenVerborgh** [RubenVerborgh/LDflex#9]
I actually changed my mind about #3: rather than implementing and returning instances of (our own or another) `Term`, a proxy path should implement the `Term` properties and methods.

So we basically need handlers for:
- [ ] `termType`
- [ ] `value`
- [ ] `equals`
- [ ] `language`
- [ ] `datatype`

and these should defer to `subject` internally.

We can implement all of them with a resolver, which I would call `DataResolver`. Then we wire it up like `new DataResolver('subject')`, and this resolver tries to resolve all properties from the internal `subject` field. Optionally, we might want to pass a second parameter to limit it to the 5 keys above.

Alternatively, given that it is only 5 fields, we might wire them up as individual handlers as `new DataHandler('subject', 'termType')`. On second thought, that might actually be my preference.

**Comment by RubenVerborgh** [comment]
@rubensworks Thoughts?

**Comment by rubensworks** [comment]
So with this you basically want to allow things like the following to be done?
```javascript
await user.friends.name.equals(someTerm);
await user.friends.name.termType;
```

In any case, if we want to pass LDflex terms to somewhere else (like Comunica),
we'd have to make sure that `await`-ing on terms directly returns a valid (synchronous) `Term`.
So the following should work as well:
```javascript
(await user.friends.name).equals(someTerm);
(await user.friends.name).termType;
```

So I'm wondering if it wouldn't make more sense to still implement our own (sync) `Term`, as we need it anyways for the second case. Your suggested `DataResolver` could them simply be a wrapper that first internally resolves the `Term`, and then returns the requested property value.

**Comment by RubenVerborgh** [comment]
Good comment, and I was wrong about the `await` (which I've now removed from the description). The idea is rather that you'd

```
const friend = await user.friends.friend
```

and that `friend` is then an LDflex object on which you can add paths (`friend.name` etc.), but also implements the regular (sync) RDF/JS interface.

**Comment by rubensworks** [comment]
Ah ok, I get the point of the `DataResolver` now.
You need this because you still want the `Term` to be an LDflex path that you can extend later.

In any case, I think that we should make sure that `DataResolver` does some internal caching. To for example avoid that multiple calls to `equals` does not result in multiple (possible expensive) operations.

So yes, 👍 to this idea.

**Comment by RubenVerborgh** [comment]
Regarding performance, I think our usage of `Proxy` makes this (relatively) slow anyway. My hunch is that the Proxy slowdown will be a magnitude larger than the actual execution of the method.

**Comment by rubensworks** [comment]
> I think our usage of Proxy makes this (relatively) slow anyway.
> My hunch is that the Proxy slowdown will be a magnitude larger than the actual execution of the method.

Hmm, that may be true.

In that case it might be good to provide a way for users to convert LDflex Terms to a more efficient representation, or at least document this.
This is in case they want to user these Terms in another library, where performance may be more important.

**Comment by RubenVerborgh** [comment]
Implemented by #13.
